src/controllers/GroupController.js

```diff
@@ -120,4 +122,91 @@ module.exports = {
       return res.status(500).send(error);
     }
   },
+  async createGroupLessons(req, res) {
+    const groupId = req.params.id;
+    try {
+      const group = await Group.findByPk(groupId);
+      if (!group) return res.status(404).send({ error: 'Group with this id was not found' });
+
+      let { startTime, endTime, createStartTime, createEndTime } = req.body;
+      if (!startTime || !endTime || !createStartTime || !createEndTime)
+        return res.status(400).send({ error: 'Invalid request parameters' });
+
+      endTime = new Date(new Date(endTime).setHours(23));
+
+      const lessons = await group.getLessons({
+        where: {
+          [Op.and]: [
+            {
+              startTime: {
+                [Op.lt]: endTime,
+              },
+            },
+            {
+              endTime: {
+                [Op.gt]: startTime,
+              },
+            },
+          ],
+        },
+      });
+
+      const range = getDatePeriod(createStartTime, createEndTime);
+
+      const existedLesson = await Lesson.findOne({
+        where: {
+          [Op.and]: [
+            {
+              groupId,
+            },
+            {
+              [Op.and]: [
+                {
+                  startTime: {
+                    [Op.lt]: createEndTime,
+                  },
+                },
+                {
+                  endTime: {
+                    [Op.gt]: createStartTime,
+                  },
+                },
+              ],
+            },
+          ],
+        },
+      });
+
+      if (existedLesson) return res.status(400).send({ error: 'Lessons already exist for this period' });
+
+      const insertBulk = () => {
+        let newLessons = [];
+        range.forEach((dd) => {
+          lessons.forEach((lesson) => {
+            if (dd.getDay() === lesson.startTime.getDay()) {
+              const newStartTime = new Date(dd.setHours(lesson.startTime.getHours()));
+              const newEndTime = new Date(dd.setHours(lesson.endTime.getHours()));
+              const newLesson = {
+                startTime: newStartTime,
+                endTime: newEndTime,
+                groupId: lesson.groupId,
+                subjectId: lesson.subjectId,
+                teacherId: lesson.teacherId,
+              };
+              newLessons.push(newLesson);
+            }
+          });
+        });
+        return newLessons;
+      }
+
+      const createdLessons = await Lesson.bulkCreate(insertBulk(), {
+        returning: true,
+      });
+      return res.send(createdLessons);
```

Comment:
Здесь думаю лучше не возвращать, т.к. может быть много данных

```suggestion
   await Lesson.bulkCreate(insertBulk());
      return res.send({ message: ' Lessons created successfully' });
```